### DISCUSSION TITLE
http to https redirect is 307, I want it to be 301, please help

### DISCUSSION THREAD

**Discussion by greggf** [vercel/community#188] [Help] [answered]
I just moved an old site to Vercel, it had never had https.   I come to find out that vercel defaults http to https via 307 temporary redirect.  This is terrible for SEO.  I need it to be a 301 redirect and can't figure out how to do this.

My site is just plain static html built with jekyll.  DNS is hosted with cloudflare.   

Can anyone help me with this please?

**Comment by fu-sen** [comment]
```
PS C:\Users\balloon> curl.exe --head http://vercel.com/
HTTP/1.1 308 Permanent Redirect
Date: Thu, 25 Mar 2021 07:31:53 GMT
Content-Type: text/plain
Connection: keep-alive
Location: https://vercel.com/
Refresh: 0;url=https://vercel.com/
server: Vercel
x-vercel-id: hnd1::2wj7l-1616657513451-da7dff087679

PS C:\Users\balloon> ▄
```

This is `308` correctly.: `308` was defined instead of `301`.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/308
https://tools.ietf.org/html/rfc7538#section-3

**Comment by fu-sen** [comment]
Also, this is a Google member's answer:
https://twitter.com/JohnMu/status/994633389356429312

**Comment by fu-sen** [comment]
This answer would be worthy of you getting:
https://stackoverflow.com/questions/42136829/whats-the-difference-between-http-301-and-308-status-codes

**Comment by vrease** [comment, marked as answer]
Hi @greggf! I have the same problem, did you find a solution? I saw in the network tab in dev tools that my site (hosted on Vercel) used a 307 from http to https. I tried the same url with `curl` and it showed a 308. Do not know why. Any problems in Search Console?

**Comment by Pieparker** [comment]
This conversation appears to have gone quiet. To keep discussions on topic and to prevent the resurrection of old topics, this discussion has been closed. If issues persist or a future reader stumbles upon this discussion and is exhibiting the same problem, please create a new discussion.
